### PR TITLE
javadoc: add page

### PR DESCRIPTION
**Pull request by einverne**
- [ ] The page (if new), does not already exist in the repo.
- [ ] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [ ] The page has 8 or fewer examples.
- [ ] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [ ] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [ ] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/javadoc.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# javadoc
+
+> Generate Java API documentation in HTML format from source code.
+> More information: <https://docs.oracle.com/javase/9/javadoc/javadoc-command.htm>.
+
+- Generate documentation for Java source code and save the result in the directory:
```
Comment (agnivade) [inline pages/common/javadoc.md]: in _a_ directory

File: pages/common/javadoc.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+`javadoc -d {{path/to/directory/}} {{path/to/java_source_code}}`
+
+- Generate documentation with specific encoding:
```
Comment (agnivade) [inline pages/common/javadoc.md]: with _a_ specific encoding

File: pages/common/javadoc.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+`javadoc -docencoding {{encoding_name}} {{java_source_code}}`
```
Comment (agnivade) [inline pages/common/javadoc.md]: Let's use UTF-8 instead of encoding_name

File: pages/common/javadoc.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+- Generate documentation exclude some packages:
```
Comment (agnivade) [inline pages/common/javadoc.md]: _excluding_ some packages

File: pages/common/javadoc.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# javadoc
+
+> Generate Java API documentation in HTML format from source code.
+> More information: <https://docs.oracle.com/javase/9/javadoc/javadoc-command.htm>.
+
+- Generate documentation for Java source code and save the result in a directory:
+
+`javadoc -d {{path/to/directory/}} {{path/to/java_source_code}}`
```
Comment (sbrl) [inline pages/common/javadoc.md]: Is `path/to/java_source_code` a file or a directory?

Comment (einverne) [reply pages/common/javadoc.md]: Both file and directory work.

File: pages/common/javadoc.md (hunk 3 above)
Comment (agnivade) [reply pages/common/javadoc.md]: @einverne - I see that you resolved this, but I don't see the change yet. Can you check ?

File: pages/common/javadoc.md (hunk 5 above)
Comment (sbrl) [reply pages/common/javadoc.md]: Ah, I see. I can't think of a better way to phrase this at the moment, so let's leave it as-is.

File: pages/common/javadoc.md (hunk 3 above)
Comment (sbrl) [reply pages/common/javadoc.md]: Also, same here with `java_source_code` -> `path/to/java_source_code`, and also the example below this one.

Comment (einverne) [reply pages/common/javadoc.md]: Ah, I forget to push..  :(

**Pull request merged**